### PR TITLE
Fix ARD product defs

### PR DESCRIPTION
**Pull request by uchchwhash**
### Reason for this pull request
The `pixel_quality` band of the S2A and S2B ARD products are now
classifications coming from the Fmask algorithm. The product
metadata that document the intended meaning of the classification
seems inaccurate. Also, the band now has a meaningful `nodata` value
that seems wrong.

### Proposed changes
Correct classification metadata and `nodata` values for the two products.

NOTE:
I got the classification data from the `.aux.xml` files that come with the
`.TIF` files as Josh Sixsmith showed me.

### CODE REVIEW

**Review by omad — DISMISSED** [review]
Great. Thanks for this update Imam.

Can you please check the values for the `contiguity` data as well please.

**Review by ASVincent — APPROVED** [review]

File: docs/config_samples/dataset_types/ga_s2_ard.yaml [hunk 1]
```diff
@@ -13,18 +13,19 @@ measurements:
 - name: 'pixel_quality'
   aliases: [mask, CFmask]
```
Comment (sixy6e) [inline docs/config_samples/dataset_types/ga_s2_ard.yaml]: CFmask should be Fmask.  The C version of Fmask was not used in the product creation.

File: docs/config_samples/dataset_types/ga_s2_ard.yaml [hunk 2]
```diff
@@ -260,18 +261,19 @@ measurements:
...
```
Comment (sixy6e) [inline docs/config_samples/dataset_types/ga_s2_ard.yaml]: CFmask should be Fmask. The C version of Fmask was not used in the product creation.

File: docs/config_samples/dataset_types/ga_s2_ard.yaml [hunk 3]
```diff
@@ -260,18 +261,19 @@ measurements:
...
   dtype: uint8
-  nodata: 255
+  nodata: 0
   units: '1'
   flags_definition: 
     cfmask:
       bits: [0,1,2,3,4,5,6,7]
       description: CFmask
```
Comment (sixy6e) [inline docs/config_samples/dataset_types/ga_s2_ard.yaml]: CFmask should be Fmask. The C version of Fmask was not used in the product creation.

File: docs/config_samples/dataset_types/ga_s2_ard.yaml [hunk 4]
```diff
@@ -13,18 +13,19 @@ measurements:
...
```
Comment (sixy6e) [inline docs/config_samples/dataset_types/ga_s2_ard.yaml]: CFmask should be Fmask. The C version of Fmask was not used in the product creation.

**Review by ASVincent — CHANGES_REQUESTED** [review]
Further changes are required for the product definition.

nir band (nir, t_nir) becomes nir1 (nir1, t_nir1)
rededge4 (rededge4, t_rededge4) becomes nir2 (nir2, t_nir2)

**Comment by uchchwhash** [comment]
Fixed. Waiting for @sixy6e to confirm the classification mapping for the `contiguous` band.

**Review by ASVincent — APPROVED** [review]

**Pull request merged**